app/workers/github_worker_base.rb

```diff
@@ -54,25 +54,18 @@ def thread_id
       return @thread_id
     end
 
-    def create_and_queue_build_task(sha:)
+    def create_and_queue_build_task(sha:, git_fork_config: nil)
       credential = self.provider_credential
       current_project = self.project
       current_sha = sha
       return unless Services.build_runner_service.find_build_runner(project_id: current_project.id, sha: current_sha).nil?
 
-      repo = GitRepo.new(
```

Comment:
don't do a big heavy thing before dispatching to a queue, instead, dispatch the initialization and then do the heavy lifting!